intellij-toml/schema/src/main/kotlin/org/toml/ide/schema/TomlJsonPsiWalker.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.toml.ide.schema
+
+import com.intellij.json.pointer.JsonPointerPosition
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.parentOfType
+import com.intellij.util.ThreeState
+import com.jetbrains.jsonSchema.extension.JsonLikePsiWalker
+import com.jetbrains.jsonSchema.extension.adapters.JsonPropertyAdapter
+import com.jetbrains.jsonSchema.extension.adapters.JsonValueAdapter
+import org.toml.lang.psi.*
+
+class TomlJsonPsiWalker private constructor() : JsonLikePsiWalker {
+    override fun isName(element: PsiElement?): ThreeState {
+        if (element is TomlKeySegment) return ThreeState.YES
+        return ThreeState.NO
+    }
+
+    override fun isPropertyWithValue(element: PsiElement): Boolean =
+        element is TomlKeyValue && element.value != null
+
+    override fun findElementToCheck(element: PsiElement): PsiElement? {
+        var current: PsiElement? = element
+
+        while (current != null && current !is PsiFile) {
+            if (current is TomlElement) return current
+            current = current.parent
+        }
+
+        return null
+    }
+
+    override fun findPosition(element: PsiElement, forceLastTransition: Boolean): JsonPointerPosition {
+        val position = JsonPointerPosition()
+        var current = element
+
+        while (current !is PsiFile) {
```

Comment:
Looks like this code doesn't take into account position inside arrays and array table